piqueserver/server.py

```diff
@@ -73,22 +73,10 @@
 
 web_client._HTTP11ClientFactory.noisy = False
 
-
-def create_path(path):
-    if path:
-        try:
-            os.makedirs(path)
-        except OSError:
-            pass
-
-
-def create_filename_path(path):
-    create_path(os.path.dirname(path))
-
-
-def open_create(filename, mode):
-    create_filename_path(filename)
-    return open(filename, mode)
+def make_parent_dirs(filename):
+    d = os.path.dirname(filename)
+    if not os.path.exists(d):
```

Comment:
It's likely not an issue here, but this can cause race conditions. It's better to try create the directory and then handle the error afterwards. This could happen e.g. if two instances are started at exactly the same time.

Reply:
Hmm true true